subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/img_mgmt.c

```diff
@@ -257,24 +257,59 @@ img_mgmt_erase(struct mgmt_ctxt *ctxt)
 {
 	struct image_version ver;
 	int rc;
+	zcbor_state_t *zsd = ctxt->cnbd->zs;
 	zcbor_state_t *zse = ctxt->cnbe->zs;
 	bool ok;
+	uint32_t image = 1;
+
+	if (!zcbor_map_start_decode(zsd)) {
+		return MGMT_ERR_EINVAL;
+	}
+
+	/* Optional image number */
+	do {
+		struct zcbor_string key;
+		static const char image_keyword[] = "image";
+
+		ok = zcbor_tstr_decode(zsd, &key);
+
+		if (ok) {
+			if (key.len == (ARRAY_SIZE(image_keyword) - 1) &&
+			    memcmp(key.value, image_keyword, ARRAY_SIZE(image_keyword) - 1) == 0) {
+				break;
+			}
+
+			ok = zcbor_any_skip(zsd, NULL);
+		}
+	} while (ok);
+
+	if (ok) {
+		ok = zcbor_uint32_decode(zsd, &image);
+		if (!ok) {
+			return MGMT_ERR_EINVAL;
+		}
+	}
+
```

Comment:
Can this be done with call to zcbor_decode_bulk with single decoding specification?

Reply:
Fixed